components/eamxx/src/physics/dp/impl/dp_iop_setinitial_impl.hpp

```diff
@@ -12,11 +12,203 @@ namespace dp {
+              auto zmask = k > thelev ? Smask(true) : tobs(k) != 0;
+              if (have_t) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  if (zmask[p]) {
+                    elem.m_forcing.m_ft(ie,i,j,k)[p] = tobs(k)[p];
+                  }
+                }
+              }
+              if (have_q) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  if (zmask[p]) {
+                    tracers.Q(ie,0,i,j,k)[p] = qobs(k)[p];
+                  }
+                }
+              }
+            });
+
+            if (have_ps) {
+              elem.m_state.m_ps_v(ie, 0, i, j) = psobs; // what is [NUM_TIME_LEVELS]?
+            }
+
+            Kokkos::parallel_for(
+              Kokkos::TeamVectorRange(team, plev_packs), [&] (int k) {
+              if (have_u) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  elem.m_state.m_v(ie, 0, 0, i, j, k)[p] = uobs(k)[p]; // [2] is 0?
+                }
+              }
+              if (have_v) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  elem.m_state.m_v(ie, 0, 1, i, j, k)[p] = vobs(k)[p]; // [2] is 1?
+                }
+              }
+              if (have_numliq) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  tracers.Q(ie, inumliq, i, j, k)[p] = numliqobs(k)[p];
+                }
+              }
+              if (have_cldliq) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  tracers.Q(ie, icldliq, i, j, k)[p] = cldliqobs(k)[p];
+                }
+              }
+              if (have_numice) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  tracers.Q(ie, inumice, i, j, k)[p] = numiceobs(k)[p];
+                }
+              }
+              if (have_cldice) {
+                vector_simd for (Int p = 0; p < Spack::n; ++p) {
+                  tracers.Q(ie, icldice, i, j, k)[p] = cldiceobs(k)[p];
+                }
+              }
+
+              // If DP-CRM mode we do NOT want to write over the dy-core vertical
+              // velocity with the large-scale one.  wfld is used in forecast.F90
+              // for the compuation of the large-scale subsidence.
+              elem.m_derived.m_omega_p(ie, i, j, k) = 0;
+            });
+          }
+        }
+      }
+      }
+    });
+  }
+
+  // If DP-CRM mode then SHOC/CLUBB needs to know about grid
+  // length size.  The calculations of this based on a sphere in the
+  // SHOC and CLUBB interefaces are not valid for a planar grid, thus
+  // save the grid length from the dycore. Note that planar dycore
+  // only supports uniform grids, thus we only save one value.
+  // Set this if it is the first time step or the first restart step
+  if ( (nstep == 0 || is_first_restart_step) && dynproc) {
+    // for (Int ie = 0; ie < nelemd; ++ie) {
```

Comment:
The f90 code here made no sense to me. It was iterating over all of dx_short, assigning (not adding) to dyn_dx_size. If that was the intent, why not just assign it to the last item?

Reply:
Agreed, the F90 makes no sense. dyn_dx_size should just be dx_short(nelemd-1)*1000

Reply:
Ah - from reading the comments, I think dyn_dx_size *was* multidimensional for global runs and got converted to a scalar for DP because (as the comments say) the planar dycore only supports uniform grids. The for loop probably made sense for global grids and should have been removed, but wasn't. I'm unclear why dx_short is still multi-dimensional in DP, but maybe it is inherently part of the dycore. If so, is dx_short(i) the same for all i? Otherwise how do we know that setting dyn_dx_size to the last instead of first entry is the way to go?